tapalcatl_server/server.go

```diff
@@ -333,15 +331,11 @@ func main() {
 	// metrics writer configuration
 	var mw metricsWriter
 	if metricsStatsdAddr != "" {
-		if metricsInterval <= 0 {
-			logger.Fatalf("ERROR: metrics interval should be greater than 0")
-		}
 		udpAddr, err := net.ResolveUDPAddr("udp4", metricsStatsdAddr)
 		if err != nil {
 			logger.Fatalf("ERROR: Invalid metricsstatsdaddr %s: %s", metricsStatsdAddr, err)
 		}
-		metricsIntervalDuration := time.Second * time.Duration(metricsInterval)
-		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, metricsIntervalDuration)
+		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, logger)
```

Comment:
I had a play around and it seems like idle goroutines and channels get garbage collected. I have to admit, I don't fully understand how the logic works around that, so I'm a little paranoid that it'll just deadlock. But that doesn't seem to be the case here.